File: apprefurbished/src/main/java/com/androidcollider/easyfin/transactions/list/TransactionsMVP.java
package com.androidcollider.easyfin.transactions.list;

import android.util.Pair;

import androidx.annotation.Nullable;

import com.androidcollider.easyfin.common.models.Transaction;
import com.androidcollider.easyfin.common.models.TransactionCategory;

import java.util.List;

import io.reactivex.rxjava3.core.Flowable;

/**
 * @author Ihor Bilous
 */

public interface TransactionsMVP {

    interface Model {

        Flowable<Pair<List<TransactionViewModel>,
                Pair<List<TransactionCategory>, List<TransactionCategory>>>> getTransactionAndTransactionCategoriesLists();

        Flowable<Transaction> getTransactionById(int id);

        Flowable<Boolean> deleteTransactionById(int id);
    }

    interface View {

        void setTransactionAndTransactionCategoriesLists(List<TransactionViewModel> transactionList,
                                                         List<TransactionCategory> transactionCategoryIncomeList,
                                                         List<TransactionCategory> transactionCategoryExpenseList);

        void goToEditTransaction(Transaction transaction);

        void deleteTransaction();
    }

    interface Presenter {

        void setView(@Nullable TransactionsMVP.View view);

        void loadData();

        void getTransactionById(int id);

        void deleteTransactionById(int id);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/transactions/list/TransactionsMVP.kt
package com.androidcollider.easyfin.transactions.list

import com.androidcollider.easyfin.common.models.Transaction
import com.androidcollider.easyfin.common.models.TransactionCategory
import io.reactivex.rxjava3.core.Single

/**
 * @author Ihor Bilous
 */
interface TransactionsMVP {
    interface Model {
        val transactionAndTransactionCategoriesLists:
                Single<Pair<List<TransactionViewModel>,
                        Pair<List<TransactionCategory>, List<TransactionCategory>>>>

        fun getTransactionById(id: Int): Single<Transaction>
        fun deleteTransactionById(id: Int): Single<Boolean>
    }

    interface View {
        fun setTransactionAndTransactionCategoriesLists(
            transactionList: List<TransactionViewModel>,
            transactionCategoryIncomeList: List<TransactionCategory>,
            transactionCategoryExpenseList: List<TransactionCategory>
        )

        fun goToEditTransaction(transaction: Transaction)
        fun deleteTransaction()
    }

    interface Presenter {
        fun setView(view: View?)
        fun loadData()
        fun getTransactionById(id: Int)
        fun deleteTransactionById(id: Int)
    }
}